### DISCUSSION TITLE
dotlottie compatibility v1 vs v2

### DISCUSSION THREAD

**Discussion by mariocampacci** [dotlottie/player-component#313] [Q&A] [answered]
Hi!

I'm trying to use this really cool lib and I noticed that the v1 is compatible with vue2 while the v2 isn't, but i cannot understand why... 
Seems related to the webpack loader not being able to parse the /dotlottie-player.mjs file.

Furthermore i haven't found nothing in the docs about compatibility stating that dotlottie v2 works only with vue3 and not with vue2. Am I using it incorrectly or are the compatibility assumptions I made correct?

Thanks!

Mario

**Comment by theashraf** [comment]
Thank you, @mariocampacci. Could you please provide us with the error message you are facing when using the dotlottie-player.mjs module?

**Reply by mariocampacci** [reply, marked as answer]
Sorry for this, found the problem. 
I'm using webpack@4 that does not support optional chaining operator parsing while webpack@5 does.
For reference in case someone is facing the same issue, installing babel and it's config solved it:
```javascript
module.exports = function (api) {
    api.cache.never()
    const presets = ['@babel/preset-env']
    return { presets }
}
```

Digging in the dotlottie repo environment I found: [@lottiefiles/dotlottie-web](https://github.com/LottieFiles/dotlottie-web/blob/main/packages/web/README.md)
@theashraf even if it still a beta should I go for it? Or stick to the @dotlottie/player-component?
Seems like the two repo have the same goal, are you switching/reimplementing this repo to @lottiefiles?
Thank you!

**Reply by theashraf** [reply]
@mariocampacci 

Great question!!

The `@dotlottie/player-component` uses `lottie-web` for rendering, which is great and works well, but it's a bit different from how things are done on `lottie-ios` and `lottie-android`, causing some rendering inconsistencies across platforms.

To ensure a uniform rendering experience everywhere, we've switched to using a new renderer named [ThorVG](https://github.com/thorvg/thorvg), which is both portable and performant.

I'd recommend trying out `@lottiefiles/dotlottie-web`. Also, check out our recent [blog](https://lottiefiles.com/blog/working-with-lottie-animations/development-update-of-dotlottie-format) post for more details on what we're up to with the dotLottie format and players. It's an exciting time to get involved!
